crates/database/src/events.rs

```diff
@@ -47,6 +47,7 @@ pub async fn last_block(ex: &mut PgConnection) -> Result<i64, sqlx::Error> {
     const QUERY: &str = "\
             SELECT GREATEST( (SELECT COALESCE(MAX(block_number), 0) FROM trades), (SELECT \
                          COALESCE(MAX(block_number), 0) FROM settlements), (SELECT \
+                         COALESCE(MAX(block_number), 0) FROM settlement_observations),(SELECT \
```

Comment:
It doesn't make sense to me that we're handling the settlement_observations table in this function. This function shouldn't touch the table.Settlement events go into the `settlements` table. Why do we need to also put them in another table? This is related to another comment where you said you had to change some fields to be non null. I don't think this is the case. You should query the settlements table joined with the observations table in `get_settlement_event_without_observation` instead.

`auction_transaction` is designed to not have coumns be null because rows only get inserted when we have the data. This PR should work the same way.

Reply:
I also mentioned that this shouldn't be here but for slightly different reasons.

Reply:
Addressed in https://github.com/cowprotocol/services/pull/1271